chains/evm/calls/transactor/monitored/monitored.go

```diff
@@ -133,52 +158,79 @@ func (t *MonitoredTransactor) Monitor(
 				for k, v := range t.pendingTxns {
 					pendingTxCopy[k] = v
 				}
+				pendingTxTraceIDCopy := make(map[common.Hash]traceapi.TraceID, len(t.pendingTxnsTrace))
+				for k, v := range t.pendingTxnsTrace {
+					pendingTxTraceIDCopy[k] = v
+				}
 				t.txLock.Unlock()
 
 				for oldHash, tx := range pendingTxCopy {
+					if time.Since(tx.submitTime) < tooNewTransaction {
+						continue
+					}
+					tID, ok := pendingTxTraceIDCopy[oldHash]
+					if ok {
+						// Creating span context with existing TraceID
+						spanCtx := traceapi.NewSpanContext(traceapi.SpanContextConfig{TraceID: tID, Remote: true})
+						ctx = traceapi.ContextWithSpanContext(ctx, spanCtx)
+					}
+					ctx, span := otel.Tracer("relayer-sygma").Start(ctx, "relayer.sygma.evm.transactor.Monitor")
+					logger := log.With().Str("dd.trace_id", span.SpanContext().TraceID().String()).Logger()
+
 					receipt, err := t.client.TransactionReceipt(context.Background(), oldHash)
 					if err == nil {
 						if receipt.Status == types.ReceiptStatusSuccessful {
-							log.Info().Uint64("nonce", tx.nonce).Msgf("Executed transaction %s with nonce %d", oldHash, tx.nonce)
+							logger.Info().Uint64("nonce", tx.nonce).Msgf("Executed transaction %s with nonce %d", oldHash, tx.nonce)
+							span.AddEvent("Executed transaction", traceapi.WithAttributes(attribute.String("tx.hash", oldHash.String()), attribute.Int64("tx.nonce", int64(tx.nonce))))
```

Comment:
I think it is nice to have all the necessary info in every particular span, when you go down the stack you are looking for the last event basically, so it would not require you to go up back to search for a necessary event. The same logic can be applied when searching a particular span, in this case, you can find all monitor events by searching with txHash.

Also if the transaction was resent with he gasPrice this TxHash will be different than the one in Executer or anywhere else higher in stack

Reply:
You have span.Status(OK, "Executed transation") just below this line.